### PR TITLE
feature/ele 540 support fetching the signing key from a jwks

### PR DESCRIPTION
**Pull request by fred-o**
- **WIP**
- **project lead is allergic to changes...**

### CODE REVIEW

**Comment by hugowetterberg** [comment]
PROJECT LEAD IS NOT ALLERGIC TO CHANGES! :wink:

**Comment by fred-o** [comment]
I blame `git yolo` :)

**Comment by fred-o** [comment]
Promise I will squash commits properly this time :)

**Comment by hugowetterberg** [comment]
Awesome! :joy: Apart from the randomly generated bad attitude the PR looks really good so far.
...I wonder how many awkward situations `git yolo` has created for its users.

File: go.sum [hunk 1]
```diff
@@ -49,6 +53,10 @@ github.com/go-test/deep v1.0.2 h1:onZX1rnHT3Wv6cqNgYyFOOlgVKJrksuCMCRvJStbMYw=
 github.com/go-test/deep v1.0.2/go.mod h1:wGDj63lr65AM2AQyKZd/NYHGb0R+1RLqB8NKt3aSFNA=
 github.com/golang-jwt/jwt/v4 v4.5.0 h1:7cYmW1XlMY7h7ii7UhUyChSgS5wUJEnm9uZVTGqOWzg=
 github.com/golang-jwt/jwt/v4 v4.5.0/go.mod h1:m21LjoU+eqJr34lmDMbreY2eSTRJ1cv77w39/MY0Ch0=
+github.com/golang-jwt/jwt/v5 v5.2.0 h1:d/ix8ftRUorsN+5eMIlF4T6J8CAt9rch3My2winC1Jw=
```
Comment (hugowetterberg) [inline go.sum:56]: Looks like you need to run a `go mod tidy`

File: jwt_test.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	token := jwt.NewWithClaims(jwt.SigningMethodES384, elephantine.JWTClaims{
+		RegisteredClaims: jwt.RegisteredClaims{
+			Issuer: "test",
+		},
+	})
+
+	ss, err := token.SignedString(jwtKey)
+	test.Must(t, err, "sign JWT token")
+
+	_, _ = parser.AuthInfoFromHeader(fmt.Sprintf("Bearer %s", ss))
```
Comment (hugowetterberg) [inline jwt_test.go:30]: What's the point of this last call?

**Review by hugowetterberg — CHANGES_REQUESTED** [review]

**Review by hugowetterberg — APPROVED** [review]

**Pull request merged**